bootstrap/service.go

```diff
@@ -176,57 +181,76 @@ func (bs bootstrapService) Add(ctx context.Context, token string, cfg Config) (C
 }
 
 func (bs bootstrapService) View(ctx context.Context, token, id string) (Config, error) {
-	owner, err := bs.identify(ctx, token)
+	user, err := bs.identify(ctx, token)
 	if err != nil {
 		return Config{}, errors.Wrap(svcerr.ErrAuthentication, err)
 	}
-	cfg, err := bs.configs.RetrieveByID(ctx, owner, id)
+	_, err = bs.authorize(ctx, user.GetDomainId(), auth.UsersKind, user.GetId(), auth.ViewPermission, auth.ThingType, id)
+	if err != nil {
+		return Config{}, err
+	}
+
+	cfg, err := bs.configs.RetrieveByID(ctx, user.GetDomainId(), id)
 	if err != nil {
 		return Config{}, errors.Wrap(svcerr.ErrViewEntity, err)
 	}
 	return cfg, nil
 }
 
 func (bs bootstrapService) Update(ctx context.Context, token string, cfg Config) error {
-	owner, err := bs.identify(ctx, token)
+	user, err := bs.identify(ctx, token)
 	if err != nil {
 		return errors.Wrap(svcerr.ErrAuthentication, err)
 	}
+	_, err = bs.authorize(ctx, user.GetDomainId(), auth.UsersKind, user.GetId(), auth.EditPermission, auth.ThingType, cfg.ThingID)
```

Comment:
What if `ThingID` is empty?

Reply:
If `ThingID` is empty then authorization will fails

Reply:
In additon, In API layer we validate the ThingID , If ThingID is empty, API will retun error